support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -323,23 +323,42 @@ object DigitalSubscriptionGiftRedemption {
     val codeValidator = new GiftCodeValidator(zuoraService)
     for {
       codeValidation <- codeValidator.getStatus(redemptionData.redemptionCode, Some(state.requestId.toString))
-      subIdUpdateAction <- codeValidation match {
-        case ValidGiftCode(subscriptionId) => Future.successful((
+      termDates <- codeValidation match {
+        case ValidGiftCode(subscriptionId) => doZuoraUpdates(
           subscriptionId,
-          zuoraService.updateSubscriptionRedemptionData(subscriptionId, state.requestId.toString, state.user.id, LocalDate.now(), _)
-        ))
-        case CodeRedeemedInThisRequest(subscriptionId) => Future.successful(subscriptionId, (_: Int) => Future.successful(UpdateRedemptionDataResponse(true)))
+          zuoraService.updateSubscriptionRedemptionData(subscriptionId, state.requestId.toString, state.user.id, LocalDate.now(), _),
+          zuoraService.setupRevenueRecognition,
+          zuoraService,
+          catalogService
+        )
+        case CodeRedeemedInThisRequest(subscriptionId) => doZuoraUpdates(
+          subscriptionId,
+          (_: Int) => Future.successful(ZuoraSuccessOrFailureResponse(success = true, None)),
+          (_: Subscription, _:TermDates) => Future.successful(()),
+          zuoraService,
+          catalogService
```

Comment:
maybe it's marginal here, but to help us focus on the business logic/reduce clutter, I'd be tempted to make `doZuoraUpdates` a method in  new `class ZuoraUpdater` and pass those services in as constructor parameters.  Then we do zuoraUpdater.doZuoraUpdates each time instead of those two lines.